tests/integration/test_scan.py

```diff
@@ -406,3 +406,43 @@ async def test_scan_sigstop(
             (30, f'Workflow not running: {name}')
             in [(level, msg) for _, level, msg in caplog.record_tuples]
         )
+
+
+@pytest.fixture
+def cylc7_run_dir(tmp_path):
+    """A run directory containing three Cylc 7 workflows."""
+    # a workflow that has not yet been run
+    # (could be run by either cylc 7 or 8 so should appear in scan results)
+    either = tmp_path / 'either'
+    either.mkdir()
+    (either / WorkflowFiles.SUITE_RC).touch()
+
+    # a Cylc 7 workflow that has been / is being run by Cylc 7
+    # (should not appear in scan results)
+    cylc7 = tmp_path / 'cylc7'
+    cylc7.mkdir()
+    (cylc7 / WorkflowFiles.SUITE_RC).touch()
+    Path(cylc7, WorkflowFiles.LOG_DIR, 'suite').mkdir(parents=True)
+    Path(cylc7, WorkflowFiles.LOG_DIR, 'suite', 'log').touch()
+
+    # a Cylc 7 workflow running under Cylc 8 in compatibility mode
+    # (should appear in scan results)
+    cylc8 = tmp_path / 'cylc8'
+    cylc8.mkdir()
+    (cylc8 / WorkflowFiles.SUITE_RC).touch()
+    Path(cylc8, WorkflowFiles.LOG_DIR, 'workflow').mkdir(parents=True)
+    Path(cylc8, WorkflowFiles.LOG_DIR, 'workflow', 'log').touch()
```

Comment:
I think we should test that a `suite.rc` workflow previously run under Cylc 7 but now running under Cylc 8 in back-compat mode should appear in the scan results. I.e.
```
~/cylc-run/my_workflow
|-- log
|   |-- suite
|   |   `-- log
|   `-- workflow
|       `-- log
`-- suite.rc
```
I have tested this manually and it does work.

Reply:
If someone has deleted the Cylc 7 DB but left the `log/suite` dir behind

Reply:
Somewhat over the top but ok, why not.